src/Domain/ZCurve.cpp

```diff
@@ -0,0 +1,91 @@
+// Distributed under the MIT License.
```

Comment:
It's worth noting that I only factored this out because (a) at the time, I thought we would have both `ElementDistribution.cpp` and `WeightedElementDistribution.cpp` that both needed it and (b) I wanted to be able to call the z-curve function from a test (on develop, it's in an anonymous namespace in `ElementDistribution.cpp` and thus has no external linkage). Given this, do you still think it's worth factoring it out, or should I just scrap the first commit altogether?

Reply:
IMO factoring out the z-curve is nice